docs/getting-started.md

```diff
@@ -0,0 +1,119 @@
+# Getting Started
+
+Welcome to the Getting Started guide for the **refarch-templates**. This guide is
+organized into sections covering **Frontend/Web Component**, **Backend/EAI**, and
+**Stack**, along with a general setup and requirements that apply to all templates.
+
+The purpose of these templates is to provide developers with an out-of-the-box
+solution for building full-stack applications.
+
+## General
+
+To use the refarch-templates, you need to have Docker installed on your
+machine. Inside the `stack` folder, you will find a `docker-compose.yml`
+file that will spin up everything needed for local development.
+
+When developing locally, the stack should always be running.
+
+### Download Templates
+
+You can download the templates or clone them directly from the GitHub
+repository.
+
+```bash
+git clone https://github.com/it-at-m/refarch-templates.git
+```
+
+Once the templates are available locally, copy the
+folders you need into your own project directory. For example, if you want
+to work with the backend, you should copy the `refarch-backend` folder.
+
+**Important**: You also need to copy the Stack folder to your project once.
+
+## Stack
+
+The stack includes:
+
+### Keycloak
+
+- **Local Instance**: Set up and run a Keycloak instance locally.
+
+### Keycloak Migration
+
+- **Migration Tool**: Utilize a migration tool that executes scripts upon startup.
+- **Script Location**: All necessary scripts are located in the
+`stack/keycloak` folder.
+Extensions to Keycloak should be implemented through these scripts.
+
+### PostgreSQL
+
+- **Local Instance**: Set up and run a PostgreSQL instance locally.
+
+### PGAdmin
+
+- **Local Instance**: Run a pre-configured instance of PgAdmin that connects to
+your local PostgreSQL database.
+
+### API Gateway
+
+- **Local Instance**: Launch the ITM API-gateway in your local environment.
+
+### AppSwitcher
+
+- **Repository**:[https://github.com/it-at-m/appswitcher-server](https://github.com/it-at-m/appswitcher-server)
+- **Configuration**: Configure the AppSwitcher within the `stack/appswitcher-server`
+folder.
+
+## Backend & EAI
+
+### Prerequisites
+
+- Java 21
+- Maven 3.9.7
+- Docker (Postgres/Keycloak)
+
+### Configure Templates
+
+After copying the templates, you will need to make a few adjustments:
+
+1. Rename the folder you copied.
+2. Change the package name from `de.muenchen.refarch` to
+   `de.muenchen.YourPackageName` (It's recommended to use an IDE for this
+   task, as it will automatically update the imports; otherwise, you will
+   need to do this manually).
+3. Inside the `pom.xml`, update the `artifactId`, `groupId`, and `name`
+   fields.
+
+After running the following command, the Backend/EAI is ready to use:
+
+```bash
+mvn install
+```
+
+### Profiles
+
+By default, the template supports two profiles:
+
+#### local
+
+- Contains the necessary configurations for running the application
+  locally, including settings for ports, SSO, and the database.
```

Comment:
Why is this needed?

Reply:
also got it from the review, that it would be nice to see why the profiles are needed and have a small explanation